src/templates_hdf5/templator_hdf5.org

```diff
@@ -1465,24 +1468,22 @@ trexio_hdf5_open_read_dset_sparse (const hid_t group_id,
       uint16_t* index = CALLOC(size_ranked, uint16_t);
       if (index == NULL) return TREXIO_ALLOCATION_FAILED;
       index_p = index;
-    } else {
-      index_p = data_sparse;
```

Comment:
Why did you remove this branch?

Reply:
Now `index_p` is initialized to `NULL` and used only if we are not using 32 bits. So below we check if `index_p` is `NULL`. If it is, we use the `index_sparse` pointer. It avoids the problem of having `index_p` that is sometimes  `const` and sometimes not. Now, `index_p` is never `const`, and we use `index_sparse` when needed.

Reply:
I think you are wrong. This modification disables read of sparse indices in 32-bits precision. The unit tests do not catch this probably because the test number of MOs is small and uses lower-bit indices.

This function is generic, that's why the input data pointer is void*. It is called 2 times: first for indices and then for values. In general, i would prefer to avoid messing up with the logic of sparse data I/O which is already quite complex.

Reply:
Below, we have:
```
  if (is_index == 1) {
    status = H5Dread(dset_id,
                     dtype,
                     memspace_id, fspace_id, H5P_DEFAULT,
                     index_p);
  } else {
    status = H5Dread(dset_id,
                     dtype,
                     memspace_id, fspace_id, H5P_DEFAULT,
                     data_sparse);
  }
```
so we use `data_sparse` if `is_index` is not 1. It seems OK to me.

Also, in my last push, I implemented tests for different numbers of MOs: 10, 20, 40, 100, 300.
They all seem to be OK.